listenerutil/parse.go

```diff
@@ -387,3 +413,122 @@ func ParseSingleIPTemplate(ipTmpl string) (string, error) {
 		return "", fmt.Errorf("multiple addresses found (%q), please configure one", out)
 	}
 }
+
+var validCustomStatusCodeCollection = []string{
+	"default",
+	"1xx",
+	"2xx",
+	"3xx",
+	"4xx",
+	"5xx",
+}
+
+const strictTransportSecurity = "max-age=31536000; includeSubDomains"
+
+// parseCustomResponseHeaders takes a raw config values for the
+// "custom_response_headers". It makes sure the config entry is passed in
+// as a map of status code to a map of header name and header values. It
+// verifies the validity of the status codes, and header values. It also
+// adds the default headers values.
+func parseCustomResponseHeaders(responseHeaders interface{}) (map[string]map[string]string, error) {
+	h := make(map[string]map[string]string)
+	// if r is nil, we still should set the default custom headers
+	if responseHeaders == nil {
+		h["default"] = map[string]string{"Strict-Transport-Security": strictTransportSecurity}
+		return h, nil
+	}
+
+	customResponseHeader, ok := responseHeaders.([]map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("response headers were not configured correctly. please make sure they're in a slice of maps")
```

Comment:
The context in which a user will see this error is if they misconfigure their HCL. We should adapt the error message. Per the [HCL definitions](https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md), I think we should use "list".
```suggestion
		return nil, fmt.Errorf("response headers were not configured correctly. Please make sure they're in a list of maps")
```